internal/pkg/service/common/httpserver/error.go

```diff
@@ -45,7 +47,11 @@ func (wr *ErrorWriter) Write(ctx context.Context, w http.ResponseWriter, err err
 }
 
 func (wr *ErrorWriter) WriteOrErr(ctx context.Context, w http.ResponseWriter, err error) error {
-	requestID, _ := ctx.Value(middleware.RequestIDCtxKey).(string)
+	// Get or generate requestID
+	requestID, ok := ctx.Value(middleware.RequestIDCtxKey).(string)
+	if !ok {
+		requestID = idgenerator.RequestID()
+	}
```

Comment:
I reused the `httpserver.ErrorWriter` for the `httpsource`, here are some small improvements.